src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  isLoader?: boolean,
+  removeTodoOnServer: (id: number) => void,
+  updateTodoOnServer: (todo: Todo, isChange?: boolean) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isLoader,
+  removeTodoOnServer,
+  updateTodoOnServer,
+}) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+  const [query, setQuery] = useState<string>(title);
+  const [hasLoader, setHasLoader] = useState(false);
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+
+  useEffect(() => {
+    setHasLoader(false);
+  }, [todo]);
+
+  const editTodo = () => {
+    const editedTodo = {
+      ...todo,
+      title: query,
+    };
+
+    if (query === '') {
+      removeTodoOnServer(id);
+    }
+
+    updateTodoOnServer(editedTodo);
+    setIsEditing(false);
+  };
+
+  const handleRemove = () => {
+    removeTodoOnServer(id);
+    setHasLoader(true);
+  };
+
+  const handleStatusChange = () => {
+    setHasLoader(true);
+    updateTodoOnServer(todo, true);
+  };
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (query !== todo.title) {
+      setHasLoader(true);
+      editTodo();
+    } else {
+      setIsEditing(false);
+    }
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setIsEditing(false);
+      setQuery(title);
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  return (
+    <div
+      className={classNames('todo', { completed: completed === true })}
```

Comment:
```suggestion
      className={classNames('todo', { completed: !!completed })}
```

OR
If completed always boolean
```suggestion
      className={classNames('todo', { completed })}
```